api/communication-api/src/main/java/com/cognifide/aet/communication/api/wrappers/SuiteRunWrapper.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * AET
+ *
+ * Copyright (C) 2013 Cognifide Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.cognifide.aet.communication.api.wrappers;
+
+import com.cognifide.aet.communication.api.metadata.RunType;
+import com.cognifide.aet.communication.api.metadata.Suite;
+
+public class SuiteRunWrapper implements Run {
```

Comment:
When you change `RunDecorator` to keep template info like `public abstract class RunDecorator<T> implements Run<T> {` you can change here and in other classes to:
`public class SuiteRunWrapper implements Run<Suite> {`

Thanks to this change you will have:

```java
  @Override
  public void setObjectToRun(Suite object) {
    this.suite = object;
  }
```